### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by katmoody** [webslides/webslides#62]
Hi! I've really appreciated finding and playing with the WebSlides project for part of an upcoming project I'm helping a friend with. As part of fully integrating it within her site, though, I've been playing with the idea to just put it together in a plugin for her WordPress site, rather than as a subdomain as I was originally intending. 

Would creating a plugin from the base code be something you guys would be interested in seeing? I would be really interested in trying to do it, even if just for this project, but wondered if having a plugin would be a neat way for the WebSlides project to spread to a different community? I would of course be including full attribution within the code base - I really just want to make it easier for WordPress users to integrate a WebSlides project, and maybe if there is interest I can add more of a UI to the editor later.

**Comment by displaynone** [comment]
Hi @katmoody

As a WordPress fanatic and as a team member of WebSlides I'll love to have a plugin that integrates WS with WP.

I also had in mind to develop a plugin, but I didn't have time yet.

Please, let us know about your plugin as soon as you have it. I also love to help you with anything you need: testing or developing.

Regards
Luis

**Comment by badcat** [comment]
I'd love to see this developed as a WP Plugin too!  I've used **[story|ftw](http://storyftw.com/)**  which I like, but the **[plugin](https://wordpress.org/plugins/storyftw/screenshots/)** hasn't been updated in a while. Would be interesting to see a WebSlides take on this.

**Comment by katmoody** [comment]
I've only worked on smaller plugins in the past, but do technical support for the BackUpWordPress plugin and have been branching out and learning more development, including more javascript, etc. I don't think the plugin itself will be an issue - what will cause me issues is trying to set up the UI so that users can more easily input webslide components. I'd want to play with whether to do that through shortcodes or not ;-) Since there's interest I will definitely keep you guys updated as soon as I have something worth looking at - and any additional notes you want to add, or features you think would be essential for a minimal viable plugin … please comment here with them. 

I'll start up a repo once I have something set up, and will invite you guys so you can give feedback and contribute further, etc. ;-)

**Comment by displaynone** [comment]
Hi  @badcat, I didn't know this plugins, thanks for the tip

@katmoody great news! I didn't think deeply about how to face the development. Shortcodes would be a good idea, they are used in page builders like Visual Composer (which I dislike) and they would help us to use modules.

**Comment by katmoody** [comment]
yes, @displaynone -- I like shortcodes, just don't like locking someone into using plugins with them, because the shortcodes basically break the functionality of the site if it's installed - potentially on important pages. It's the biggest reason I don't like some page builders. 

But I think shortcodes might be the most extensible option available to us in the short-term. And it would use the least amount of resources to do as long as the shortcode outputs clean code.

**Comment by displaynone** [comment]
I feel the same about shortcodes @katmoody, if you deactivate the plugin the content will be a mesh.

What do you think about hierarchical content? the parent CPT will be the  slideshow container, perhaps with some options like common header or footer. The child CPT will be each slide.

**Comment by jlantunez** [comment]
Hi guys,

What do you think about something like this? Will it be easier to understand for a beginner?

![webslides-wordpress-editor](https://cloud.githubusercontent.com/assets/2033183/24208344/b12adf6a-0f23-11e7-8fda-acc9c6d872f5.png)

**Comment by displaynone** [comment]
Love it @jlantunez

**Comment by jlantunez** [comment]
I forgot to add "webslides/slideshow" in the format box

**Comment by katmoody** [comment]
That is very close to what I was originally thinking about @jlantunez and @displaynone I haven't done nested CPTs before but I'm intrigued by the idea. 

Essentially you're looking a slideshow/even just a landing page. What are the key questions a user will have: 

- will this display like a traditional WP page or have a full-width, no header, no footer custom template? (well the beginner might instead as if they can put it in a page or post, which I think it would be nice to be able to offer, but I envision the full-screen experience to replicate the WebSlides experience)
- Will they decide if they want a header (ie logo and/or navigationi) and a footer (ie social links, etc)? Thinking in terms of the newbie, that's how they would see the header and footer of the slide(s) as well. Does it make sense to tie that to a custom header/footer assigned to the CPT instead of trying to nest one CPT inside another? 
- how do I use it/publish a slideshow (or single slide)

Answers to these questions would help determine how we set it up - nested CPTs versus custom header/footer/CPT templates. How would we build the UI so that it's …

1. Not overwhelming to the newbie
2. able to be easily extended by someone who knows Webslides
3. Preferably modular - a core plugin with options you can enable (also a way for you guys to potentially consider a premium level down the road?). 

This is a brainstorm so ignore anything stupid-sounding, lol

**Comment by katmoody** [comment]
The more I think about it the more I like the nested CPTs better - it's modular, can be added to a post or page and displayed inside the content area OR used as a standalone full-screen slideshow.

I'm thinking how easy it would be to create a landing page or coming soon page on the fly by making it a single slide … easily mobile-friendly, quick to build and versatile. I see more opportunity for being able to use it as a full-site tool that's easy for newbies AND developers to use.

**Comment by jlantunez** [comment]
@katmoody  We're on the same page. An easy, fast, small, and powerful solution for building presentations, landings, portfolios, long-forms...

**Comment by displaynone** [comment]
@katmoody, I think that using nested CPTs, shortcodes will not be necessary. Perhaps we can add some TinyMCE buttons for the webslides styles, so user can really see what is doing, a really WYSIWYG editor.

No stupid-sounding at all ;-)

**Comment by katmoody** [comment]
That might be a nice way to handle it, even if it is for a future release - to eventually have them all show up easily in the WYSIWYG editor. If we have the styles updated for the editor, as well, we can have it mimic the actual display of some modules/customizations.

**Comment by katmoody** [comment]
I'm looking at #63 to see whether that would be a useful idea for a clickable resource page included with the plugin eventually. I'd like to prioritize components for adding to the menu as well so we can have the list to use for future updates as well. ;-)

**Comment by lsalsich** [comment]
Hey all, wondering if this idea went any further? Thanks!

**Comment by jlantunez** [comment]
Nop :(

**Comment by katmoody** [comment]
Sorry about that - ended up in a busy season with personal and professional commitments. Still very interested though ;-)

**Comment by stracker-phil** [comment]
Great to see that some other people also want to have Webslides in WordPress :-)

As the new Block-Editor in WordPress (Gutenberg) provides a great UI that can be used to design slides, I have started a free WordPress plugin based on the Block-Editor.

The plugin is available here: https://wordpress.org/plugins/webslides-presentations/
At the moment it only has headlines and paragraphs. I am working on new elements and customizations during my spare time, so some new features will be added in the near future!

Philipp

**Comment by displaynone** [comment]
Thanks Philipp, looks great

**Comment by jlantunez** [comment]
Thanks @stracker-phil. You made my day!

**Comment by katmoody** [comment]
That's awesome @stracker-phil !!!! I haven't had the time to work on it for a long time but still loved the idea!
